src/script/auth/page/Login.tsx

```diff
@@ -214,47 +252,80 @@ const Login = ({
               centerText
               style={{display: 'flex', flexDirection: 'column', height: 428, justifyContent: 'space-between'}}
             >
-              <div>
-                <H1 center>{_(loginStrings.headline)}</H1>
-                <Muted>{_(loginStrings.subhead)}</Muted>
-                <Form style={{marginTop: 30}} data-uie-name="login">
-                  <LoginForm isFetching={isFetching} onSubmit={handleSubmit} />
-                  {validationErrors.length ? (
-                    parseValidationErrors(validationErrors)
-                  ) : loginError ? (
-                    parseError(loginError)
-                  ) : (
-                    <div style={{marginTop: '4px'}}>&nbsp;</div>
-                  )}
-                  {!Runtime.isDesktopApp() && (
-                    <Checkbox
-                      tabIndex={3}
-                      onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
-                        pushLoginData({clientType: event.target.checked ? ClientType.TEMPORARY : ClientType.PERMANENT});
-                      }}
-                      checked={loginData.clientType === ClientType.TEMPORARY}
-                      data-uie-name="enter-public-computer-sign-in"
-                      style={{justifyContent: 'center', marginTop: '12px'}}
-                    >
-                      <CheckboxLabel>{_(loginStrings.publicComputer)}</CheckboxLabel>
-                    </Checkbox>
-                  )}
-                </Form>
-              </div>
-              <Columns>
-                <Column>
-                  <Link onClick={forgotPassword} data-uie-name="go-forgot-password">
-                    {_(loginStrings.forgotPassword)}
-                  </Link>
-                </Column>
-                {Config.getConfig().FEATURE.ENABLE_PHONE_LOGIN && (
-                  <Column>
-                    <Link onClick={() => history.push(ROUTE.LOGIN_PHONE)} data-uie-name="go-sign-in-phone">
-                      {_(loginStrings.phoneLogin)}
-                    </Link>
-                  </Column>
-                )}
-              </Columns>
+              {twoFactorLoginData && (
+                <div>
+                  <H2 center>{_(loginStrings.twoFactorLoginTitle)}</H2>
+                  <Text>{_(loginStrings.twoFactorLoginSubHead, {email: twoFactorLoginData.email})}</Text>
+                  <Label markInvalid={twoFactorSubmitFailed}>
+                    <CodeInput
+                      autoFocus
+                      style={{marginTop: 60}}
+                      onCodeComplete={submitTwoFactorLogin}
+                      data-uie-name="enter-code"
+                      markInvalid={twoFactorSubmitFailed}
+                    />
+                    {twoFactorSubmitFailed && (
+                      <div>{_(errorHandlerStrings[BackendError.LABEL.CODE_AUTHENTICATION_FAILED])}</div>
```

Comment:
i think i prefer to set this more dynamically in the error handling (based off the received error label) than to display it statically only in this case.

Reply:
https://github.com/wireapp/wire-webapp/pull/12681/commits/53093655850a363db3e6e280bbdd792d4afd993b